### PR TITLE
Type Markdown stuff better

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: shared/common-adapters/markdown/react.js [hunk 1]
```diff
@@ -334,12 +335,19 @@ const reactComponentsForMarkdownType = {
     )
   },
   serviceDecoration: (node, output, state) => {
+    const {markdownMeta} = state
+    if (!markdownMeta) {
+      throw new Error('markdownMeta unexpectedly empty')
+    }
+    // $FlowIssue Don't know what's going on here.
+    const {message} = markdownMeta
```
Comment (akalin-keybase) [inline shared/common-adapters/markdown/react.js]: anyone know what's going on here? `message` isn't a key in `markdownMeta` (which itself may be null)

**Comment by akalin-keybase** [comment]
Split off from https://github.com/keybase/client/pull/15295

File: shared/common-adapters/markdown/react.js (hunk 1 above)
Comment (mmaxim) [reply shared/common-adapters/markdown/react.js]: It is, my guess is this PR breaks in-chat wallet sends.

Comment (akalin-keybase) [reply shared/common-adapters/markdown/react.js]: The code doesn't actually change any logic (it just silences the flow errors), but I'll look at the in-chat wallet send flow. Thanks!

File: shared/common-adapters/markdown/index.js.flow [hunk 2]
```diff
@@ -23,12 +22,23 @@ export type MarkdownCreateComponent = (
...
-export type MarkdownMeta = {
-  mentionsAt?: MentionsAt,
-  mentionsChannelName?: MentionsChannelName,
-  mentionsChannel?: MentionsChannel,
+export type MarkdownMeta = {|
+  mentionsAt: MentionsAt,
+  mentionsChannelName: MentionsChannelName,
+  mentionsChannel: MentionsChannel,
   message: MessageText,
-  onOpenInFilesTab?: Path => void,
```
Comment (akalin-keybase) [inline shared/common-adapters/markdown/index.js.flow]: unused afaict

**Pull request merged**